### ISSUE TITLE
Add Map-Odom Localization

### ISSUE THREAD

**Issue by amock** [naturerobots/mesh_navigation_tutorials#4]
This is a TODO issue for adding examples on how to get a map-to-odom localization. The current map-to-odom localization is provided by Gazebo. Since we cannot get the Gazebo localization in the real world, this is a necessary and missing component to apply mesh navigation to a real robot. 

Some approaches that came to my mind:
1. MICP-L from the [RMCL](https://github.com/uos/rmcl) project. This should be not too hard to do, as it was successfully done for the ROS 1 version here: https://github.com/uos/rmcl/issues/3
2. Instructions on how to convert meshes to other formats where the SOTA of prior map localization is wider. 
    - e.g. in the MICP-L paper (https://kbs.informatik.uos.de/publication/iros24_amock/), I sampled a point cloud from the mesh map using CloudCompare and made some [modifications](https://github.com/amock/kiss-icp/tree/micp-experiments) on [Kiss-ICP](https://github.com/PRBonn/kiss-icp) so that it could operate solely on prior maps.
3. Something GPS-based for outdoor scenarios. Provide instructions on how to geo-reference a map. 

I will care about point (1.) soon. However, if someone is faster than me feel free to contact me. Note: I would still like to keep the Gazebo localization as default since it makes life easier to develop navigation algorithms.

**Comment by mathew4STAR** [comment]
Hey, method 1 doesn't seem to work for me(I believe it is integrated fully). I installed rmcl and all its dependencies.But when I run it with the localization_type arguments set to rmcl_micpl I get a rmcl not found error. So I changed rmcl to rmcl_ros in line 136 of the mesh_navigation_tutorials_launch.py which did seem to fix the issue as rmcl is now launching properly.
But I am still not getting a map frame. Rmcl launches and I get the standard output but map->odom frame is not being published though, even though rmcl says it is providing odom -> map frame. I tried checking the tf tree and their is indeed no map frame.

**Comment by amock** [comment]
We integrated it once and it also worked pretty well. However, we wanted to test it first on different computers before releasing it with the tutorials here; along with documentation how to get it running. That's why it is mentioned here.

But you are on the right way. Since MICP-L is a method for pose tracking, you'll need an initial pose estimate. You can use the pose guess tool from RViz for that. Or better, you can use the pose guess tool from mesh_tools that draws a pose directly on the mesh map. When done, you should see lidar data in the map frame of RViz.

If that's not working, you may try RMCL alone first using the instructions from here: https://github.com/amock/rmcl_examples . If you have issues with it, feel free to open Issues there.

**Comment by yashbhaskar** [comment]
I'm using a custom terrain mesh made in Blender (exported as .dae/.ply) and it loads fine in Gazebo Fortress, but in RViz the mesh doesn’t show at all because the TF is broken — the map frame is missing and I keep getting Could not transform from [map] to [base_footprint]. The Mesh Map stays empty and RobotModel gives TF warnings. It looks like the launch file doesn’t publish any map frame or static transform when using custom meshes. Can you confirm if I need to manually add a map/TF setup, or if the tutorial is missing a step for custom mesh worlds?

**Comment by amock** [comment]
No, I cannot confirm that. If you set up your world like the ones in the tutorials, the ground truth localization (default) should work out of the box.

There could be many other reasons why your setup doesn’t work, but that is likely out of scope for this issue.

If you feel that any topics are missing from the tutorials, feel free to open another issue so that people can help fill the gap.
